### PR TITLE
ci: reject changesets for private packages

### PR DESCRIPTION
**Pull request by Dugyu**
## Motivation

Private workspace packages are excluded from Changesets versioning and tagging. Changesets targeting these packages therefore create misleading release metadata that cannot result in a release.

This PR adds a CI gate to reject these invalid changesets before they are merged.

## Changes

- Collect private package names from the current pnpm workspace.
- Scan `.changeset/*.md` frontmatter for private package targets.
- Report the affected changeset files and private package names.
- Run the policy check through `check:all` and PR CI.
- Extract the policy decision into a pure helper for deterministic testing.
- Add tests covering:
  - mixed public and private targets
  - target ordering
  - multiple private targets
  - invalid private workspace metadata

## Verification

- `pnpm test:changesets-private`
- `pnpm check:changesets-private`
- `pnpm check:all`
- `pnpm turbo build`

- Enforce a policy that rejects Changesets targeting private workspace packages.
- Report affected Changeset files and private package names.
- Integrate validation with `check:all` and PR CI.
- Test target ordering, multiple targets, and invalid workspace metadata.

### CODE REVIEW

File: tools/scripts/check-changeset-private-targets.mjs [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+export function getChangesetPackageNames(changesetText) {
+  const frontmatter = getFrontmatter(changesetText)
+
+  if (!frontmatter) {
+    return []
+  }
+
+  return frontmatter.map(line => parsePackageName(line)).filter(Boolean)
```
[the remark this answers is not in the document]
Comment (Dugyu) [reply tools/scripts/check-changeset-private-targets.mjs]: @coderabbitai  Addressed your review feedback on frontmatter parsing.

- Replaced the previous line-based parsing with `@changesets/parse@0.4.3`, and now use the parser’s `releases` output for target detection.
- This fixes the scoped-name edge case where YAML flow mappings (e.g. `{ "@lynx-js/private": patch }`) were not parsed correctly and could bypass the private-target gate.
- Added a regression test covering that flow-mapping release format.

**Review by gaoachao — APPROVED** [review]
